### PR TITLE
Update quantile

### PR DESCRIPTION
**Pull request by smishr**
See #87 for details. Attempting to fix the issue then add testing suite for quantile

### CODE REVIEW

File: src/mean.jl [hunk 1]
```diff
@@ -95,8 +95,8 @@ end
 
 Estimate the subpopulation mean of a variable `x`.
 
-The calculations were done according to the book [Calibration Estimators in Survey Sampling](https://www.tandfonline.com/doi/abs/10.1080/01621459.1992.10475217)
-by Jean-Claude Deville and Carl-Erik Sarndal.
+The calculations were done according to the book [Model-Assisted Survey Sampling](https://link.springer.com/book/9780387406206)
```
Comment (iuliadmtru) [inline src/mean.jl:98]: This is not an update on `quantile`. It should be in a different PR.

File: src/Survey.jl [hunk 2]
```diff
@@ -2,6 +2,7 @@ module Survey
 
 using DataFrames
 using Statistics
+import Statistics: quantile
```
Comment (iuliadmtru) [inline src/Survey.jl:5]: Doesn't this conflict with the import on line 7?

File: src/quantile.jl [hunk 3]
```diff
@@ -1,6 +1,14 @@
 """
-    quantile(var, design, q)
-Estimate quantiles for `SurveyDesign`s.
+    quantile(var, design, q; kwargs...)
```
Comment (iuliadmtru) [inline src/quantile.jl:2]:
```suggestion
    quantile(var, design, q; kwargs...)
    
```

File: src/quantile.jl [hunk 4]
```diff
@@ -13,27 +21,33 @@ julia> quantile(:enroll, srs, 0.5)
...
-    rename!(df, :tmp => Symbol(string(q) .* "th percentile"))
-    return df
-end
-
-# Inner method for `by`
-function quantile(x, w, _, q)
-    df = DataFrame(tmp = Statistics.quantile(Float32.(x), weights(w), q))
-    rename!(df, :tmp => Symbol(string(q) .* "th percentile"))
-
+    df = DataFrame(qth_quantile = q, quantile = Statistics.quantile(Float32.(x), weights(w), q))
```
Comment (ayushpatnaikgit) [inline src/quantile.jl:51]:
```suggestion
    df = DataFrame(qth_quantile = q, quantile = Statistics.quantile(Float32.(x), weights(w), q; kwargs...))
```

File: src/quantile.jl [hunk 5]
```diff
@@ -13,27 +21,33 @@ julia> quantile(:enroll, srs, 0.5)
...
+function quantile(var::Symbol, design::SimpleRandomSample, q::Union{<:Real,Vector{<:Real}}; alpha::Real=1.0, beta::Real=alpha, kwargs...)
     x = design.data[!, var]
-    df = DataFrame(tmp = Statistics.quantile(Float32.(x), q; kwargs...))
-    rename!(df, :tmp => Symbol(string(q) .* "th percentile"))
+    df = DataFrame(qth_quantile = q, quantile = Statistics.quantile(Float32.(x), q; kwargs...))
     return df
 end
 
-function quantile(var, design::StratifiedSample, q)
+function quantile(var::Symbol, design::StratifiedSample, q::Union{<:Real,Vector{<:Real}}; kwargs...)
```
Comment (iuliadmtru) [inline src/quantile.jl:48]:
```suggestion
function quantile(var::Symbol, design::StratifiedSample, q::Union{Real, Vector{<:Real}}; kwargs...)
```

File: src/quantile.jl [hunk 6]
```diff
@@ -13,27 +21,33 @@ julia> quantile(:enroll, srs, 0.5)
...
+julia> strat = load_data("apistrat");
+
+julia> dstrat = StratifiedSample(strat, :stype; popsize=:fpc);
+
+julia> quantile(:enroll, dstrat, [0.1,0.2,0.5,0.75,0.95])
+
 ```
 """
-function quantile(var, design::SimpleRandomSample, q; kwargs...)
+function quantile(var::Symbol, design::SimpleRandomSample, q::Union{<:Real,Vector{<:Real}}; alpha::Real=1.0, beta::Real=alpha, kwargs...)
```
Comment (iuliadmtru) [inline src/quantile.jl:42]:
```suggestion
function quantile(var::Symbol, design::SimpleRandomSample, q::Union{Real, Vector{<:Real}}; alpha::Real=1.0, beta::Real=alpha, kwargs...)
```

File: src/Survey.jl (hunk 2 above)
Comment (ayushpatnaikgit) [inline src/Survey.jl:5]: We are already importing quantile in line 7.

File: src/quantile.jl [hunk 7]
```diff
@@ -13,27 +21,33 @@ julia> quantile(:enroll, srs, 0.5)
...
     x = design.data[!, var]
-    df = DataFrame(tmp = Statistics.quantile(Float32.(x), q; kwargs...))
-    rename!(df, :tmp => Symbol(string(q) .* "th percentile"))
+    df = DataFrame(qth_quantile = q, quantile = Statistics.quantile(Float32.(x), q; kwargs...))
```
Comment (iuliadmtru) [inline src/quantile.jl:44]:
```suggestion
    df = DataFrame(qth_quantile=q, quantile=Statistics.quantile(Float32.(x), q; kwargs...))
```

File: src/quantile.jl (hunk 4 above)
Comment (iuliadmtru) [inline src/quantile.jl:51]:
```suggestion
    df = DataFrame(qth_quantile=q, quantile=Statistics.quantile(Float32.(x), weights(w), q))
```

Comment (smishr) [reply src/quantile.jl]: i have changed this in later commit

File: src/Survey.jl (hunk 2 above)
Comment (smishr) [reply src/Survey.jl:5]: Statistics and Statsbase have different definitions of `quantile` fn

File: src/mean.jl (hunk 1 above)
Comment (smishr) [reply src/mean.jl:98]: Okay i think it was previously committed/stashed in design_update branch.

File: src/Survey.jl (hunk 2 above)
Comment (smishr) [reply src/Survey.jl:5]: I havent received any error messages, but am still looking into it.

File: src/quantile.jl (hunk 3 above)
Comment (smishr) [reply src/quantile.jl]: I have updated this. To keep with `Statistics` and `Statsbase` syntax, we should use `p` instead of `q` for the vector of probabilities

File: src/quantile.jl (hunk 5 above)
Comment (smishr) [reply src/quantile.jl]: this has been added

**Review by smishr — COMMENTED** [review]
ready to merge

File: src/quantile.jl (hunk 7 above)
Comment (smishr) [reply src/quantile.jl]: Julia formatter ran

File: src/quantile.jl (hunk 6 above)
Comment (smishr) [reply src/quantile.jl]: Fixed
